ci-operator/step-registry/enable-qe-catalogsource/stage/enable-qe-catalogsource-stage-commands.sh

```diff
@@ -0,0 +1,191 @@
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+function update_global_auth () {
+  # get the current global auth
+  run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+  if [[ $ret -ne 0 ]]; then
+      echo "!!! fail to get the cluster global auth."
+      return 1
+  fi
+ # run_command "cat /tmp/.dockerconfigjson | jq"
+
+  # replace all global auth with the QE's
+  # new_dockerconfig="/var/run/vault/image-registry/qe_dockerconfigjson"
+
+  # only add quay.io/openshift-qe-optional-operators auth to the global auth
+  new_dockerconfig="/tmp/new-dockerconfigjson"
+  # qe_registry_auth=$(cat "/var/run/vault/mirror-registry/qe_optional.json" | jq -r '.auths."quay.io/openshift-qe-optional-operators".auth')
+  optional_auth_user=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.user')
+  optional_auth_password=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.password')
+  qe_registry_auth=`echo -n "${optional_auth_user}:${optional_auth_password}" | base64 -w 0`
+
+  reg_brew_user=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.user')
+  reg_brew_password=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.password')
+  brew_registry_auth=`echo -n "${reg_brew_user}:${reg_brew_password}" | base64 -w 0`
+  jq --argjson a "{\"brew.registry.redhat.io\": {\"auth\": \"${brew_registry_auth}\", \"email\":\"jiazha@redhat.com\"},\"quay.io/openshift-qe-optional-operators\": {\"auth\": \"${qe_registry_auth}\", \"email\":\"jiazha@redhat.com\"}}" '.auths |= . + $a' "/tmp/.dockerconfigjson" > ${new_dockerconfig}
+
+ # run_command "cat ${new_dockerconfig} | jq"
+
+  # update global auth
+  ret=0
+  run_command "oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=${new_dockerconfig}" || ret=$?
+  if [[ $ret -eq 0 ]]; then
+      echo "update the cluster global auth successfully."
+  else
+      echo "!!! fail to add QE optional registry auth, retry and enable log..."
+      sleep 1
+      ret=0
+      run_command "oc --loglevel=10 set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=${new_dockerconfig}" || ret=$?
+      if [[ $ret -eq 0 ]]; then
+        echo "update the cluster global auth successfully after retry."
+      else
+        echo "!!! still fail to add QE optional registry auth after retry"
+        return 1
+      fi
+  fi
+}
+
+function delete_resources() {
+  #Delete existing ICSP and CatalogSource
+  echo "Deleting any existing ICSP and CatalogSource"
+  oc delete imagecontentsourcepolicies brew-registry --ignore-not-found
+  oc delete catalogsource qe-app-registry -n openshift-marketplace --ignore-not-found
+}
+
+# create ICSP for connected env.
+function create_icsp_connected () {
```

Comment:
This `ICSP` is incorrect for the `stage`, and we can get it dynamically, such as: https://github.com/openshift/release/blob/master/ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh#L111-L132

Reply:
@jianzhangbjz  Thanks for advice, will update it later.